### PR TITLE
Fix #1461: Implement proper sanitization of credential ID

### LINKED ISSUE #1461: Implement proper sanitization of credential ID

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

File: powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/service/AssertionService.java [hunk 1]
```diff
@@ -95,11 +99,41 @@ public AssertionVerificationResponse authenticate(AssertionVerificationRequest r
...
+            try {
+                logger.debug("Looking up authenticator for credential ID: {}, application ID: {}", credentialId, applicationId);
+                authenticatorDetail = getAuthenticatorDetail(credentialId, applicationId);
+                logger.info("Found authenticator with ID: {}, for credential ID: {}, application ID: {}", authenticatorDetail.getActivationId(), credentialId, applicationId);
+            } catch (Fido2AuthenticationFailedException ex) {
+                logger.debug("Authenticator lookup failed, trying find trimmed version.");
+                // Try to find trimmed credential ID
+                final byte[] credentialIdBytes = Base64.getDecoder().decode(credentialId);
+                if (credentialIdBytes.length > 32) {
+                    String credentialIdTrimmed = Base64.getEncoder().encodeToString(Arrays.copyOfRange(credentialIdBytes, 0, 32));
```
Comment (banterCZ) [inline powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/service/AssertionService.java:118]: May be _final_
```suggestion
                   final String credentialIdTrimmed = Base64.getEncoder().encodeToString(Arrays.copyOfRange(credentialIdBytes, 0, 32));
```

File: powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/service/AssertionService.java [hunk 2]
```diff
@@ -95,11 +99,41 @@ public AssertionVerificationResponse authenticate(AssertionVerificationRequest r
...
+                    logger.debug("Looking up authenticator for trimmed credential ID: {}, application ID: {}", credentialIdTrimmed, applicationId);
+                    authenticatorDetail = getAuthenticatorDetail(credentialIdTrimmed, applicationId);
+                    // Check if trimming is supported
```
Comment (banterCZ) [inline powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/service/AssertionService.java:121]: I suggest extracting methods instead of commenting.

**Review by banterCZ — APPROVED** [review]
Looks good. The test is failing even in the develop branch, pipelines are somehow broken, because locally runs without problem.

**Review by romanstrobl — DISMISSED** [review]
Looks OK.

**Pull request merged**